src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/TermQueryExplainIT.java

```diff
@@ -102,19 +100,28 @@ public void testNonResolvingIndexPatternWithNonExistingIndex() throws IOExceptio
     }
 
     @Test
-    public void testNonIdenticalMappings() throws IOException {
+    public void testNonCompatibleMappings() throws IOException {
         try {
-            explainQuery(String.format(Locale.ROOT, "SELECT firstname, birthdate FROM %s, %s " +
-                            "WHERE firstname = 'Leo' OR male = 'true'",
-                    TEST_INDEX_BANK, TEST_INDEX_ONLINE));
+            explainQuery(String.format(Locale.ROOT, "SELECT * FROM %s, %s ",
+                    TEST_INDEX_DOG, TEST_INDEX_DOG2));
         } catch (ResponseException e) {
             assertThat(e.getResponse().getStatusLine().getStatusCode(), equalTo(RestStatus.BAD_REQUEST.getStatus()));
             final String entity = TestUtils.getResponseBody(e.getResponse());
-            assertThat(entity, containsString("When using multiple indices, the mappings must be identical"));
+            assertThat(entity, containsString("dog_name"));
+            assertThat(entity, containsString("{type:text,fields:{keyword:{type:keyword,ignore_above:256}}}"));
+            assertThat(entity, containsString("{type:text,fielddata:true}"));
             assertThat(entity, containsString("\"type\": \"VerificationException\""));
         }
     }
 
+    @Test
+    public void testCompatibleMappings() throws IOException {
```

Comment:
the name is misleading, since we don't check compatibility but equality, I'd rename this to `testEqualFieldMappings` or something alike. When that TODO is resolved and we really check compatibility, then we'll need to add more tests for that.

Reply:
We are not checking equality here. We are checking if the mappings for the same field name are equal or one index has extra mapping not present in others, we should not throw an exception. Does `testMappingsNotDifferent` sounds good?

Reply:
I think that's what the code here does - check that for the relevant fields the mappings are equal. As I mentioned, the //TODO comment you put on TermFieldRewriter.java#244 is to support the compatibility, but with current change it's just equality. I just want the test name to reflect that.

Reply:
I misunderstood you. I thought you want sth like `testIdenticalMapping` which we already have.